vortex/src/main/java/org/apache/iceberg/vortex/ConvertFilterToVortex.java

```diff
@@ -107,7 +107,11 @@ public Expression and(Expression leftResult, Expression rightResult) {
 
   @Override
   public Expression or(Expression leftResult, Expression rightResult) {
-    return Binary.or(leftResult, rightResult);
+    if (leftResult == UnconvertibleExpr.INSTANCE || rightResult == UnconvertibleExpr.INSTANCE) {
```

Comment:
@a10y, thanks for the prompt update! 
There are no unit test cases included in this PR. What are your thoughts or plan for adding tests to cover the conversion logic?

Reply:
Yes that is a good question. Unit tests have limited value here since expressions can be combined and cause issues (like they did here). I'll add some in a follow up, in the limit perhaps a property based test framework might give better coverage.